framework/behaviors/SluggableBehavior.php

```diff
@@ -171,6 +171,10 @@ protected function isNewSlugNeeded()
             return false;
         }
 
+        if ($this->attribute === null) {
```

Comment:
this feels a bit magical and is taking too much time to understand. Can you support this with a comment to explain why true is to be returned when attribute is null?